vnext/ReactUWP/TurboModule/TurboModuleUtils.cpp

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include <ReactUWPABI.h>
+#include <TurboModuleUtils.h>
+#include "../ABI/Instance_rt.h"
+
+namespace react::uwp {
+
+std::shared_ptr<facebook::react::Instance> __cdecl GetFacebookReactInstance(
+    const winrt::react::uwp::Instance &uwpInstance) {
+  winrt::com_ptr<ABI::react::uwp::IInstance> abiIInstance;
+  winrt::copy_to_abi(uwpInstance, *abiIInstance.put_void());
```

Comment:
Please don't use abi interface anymore.

I didn't do the testing, but I think you should use winrt::get_self.

if (auto self = winrt::get_self<::Instance>(uwpInstance)) {
         return self->GetOmmerOmstamce()'
}

Reply:
I've tested it works, and @acoates-ms suggests me to do this. May I know the reason please?

Reply:
Meanwhile I will update it to use get_self and see if it works.

Reply:
I'll defer to CanHua about cppwinrt usage.  My knowledge of it is limited. (and probably largely out of date)

Reply:
c++/winrt has a lot of benefit over WRL and we don't need to handle pointer and check hr code anymore.
currently ABI header(idl files in ReactUWP\ABI) and cppwinrt headers co-exists in react-uwp project.
Ideally we should only use c++/winrt in ReactUWP project, and migrate all [wrl to c++/winrt](https://docs.microsoft.com/en-us/windows/uwp/cpp-and-winrt-apis/move-to-winrt-from-wrl).

I hacked the project to generate ABI/Instance_rt.h, but you expect there is no ABI/Instance_rt.h in the very near future.
```
    <Midl Include="ABI\idl\ReactUWPABI.idl">
      <!-- Enable Windows Runtime ABI header files. when /reference is passed to midl.exe, the ABI head includes headers like Windows.AI.MachineLearning.MachineLearningContract.h 
      which doesn't exist actually in SDK. To workaround the problem, DisableReferences is enabled to disable /reference.      
       -->
      <DisableReferences>true</DisableReferences>
      <HeaderFileName>$(IdlHeaderDirectory)\%(Filename).h</HeaderFileName>
      <MetadataFileName>$(UnmergedWinmdDirectory)\%(Filename).winmd</MetadataFileName>
      <AdditionalOptions>
      </AdditionalOptions>
    </Midl>
```

Reply:
Tried `get_self<ABI::react::uwp::Instance>` but it gives me an error
```
'abi_GetTrustLevel': is not a member of 'ABI::react::uwp::Instance'
```

And I failed to find any sub class of this class. Do you know why this error get prompted? Is there any missing piece for this implementation class to be a C++/WinRT qualified class? Or do you think the old way is acceptable?

Reply:
@licanhua

Reply:
Look like we need to convert Instance from WRL to cppwinrt first. so you can ignore it.